docs/REST/authentication.md

```diff
@@ -348,4 +348,47 @@ private static String buildContentHash(HttpUriRequest request) throws IOExceptio
 
     byte[] digest = new DigestUtils(SHA_256).digest(content);
     return Base64.getEncoder().encodeToString(digest);
-}
\ No newline at end of file
+}
+```
+### Golang
+```golang
+import (
+    "crypto/hmac"
+    "crypto/sha256"
+    "encoding/base64"
+    "fmt"
+    "io/ioutil"
+    "net/http"
+    "strings"
+    "time"
+)
+
+func getContentHashBase64(content string) string {
+    hasher := sha256.New()
+    hasher.Write([]byte(content))
+    return base64.StdEncoding.EncodeToString(hasher.Sum(nil))
+}
+
+func signRequest(content string, key []byte) string {
```

Comment:
Technically, this is not signing a request. It's just calculating hmac of a given string. Maybe `getHmac` is better?

Can these helper functions be put to the end of the code or do they have to be declared before called in go?